src/Integration.Vsix.UnitTests/CFamily/MessageHandlerTests.cs

```diff
@@ -0,0 +1,128 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using SonarLint.VisualStudio.Core.Analysis;
+using SonarLint.VisualStudio.Core.CFamily;
+using SonarLint.VisualStudio.Integration.Vsix.CFamily;
+
+namespace SonarLint.VisualStudio.Integration.UnitTests.CFamily
+{
+    [TestClass]
+    public class MessageHandlerTests
+    {
+        [TestMethod]
+        public void TestIsIssueForActiveRule()
+        {
+            var rulesConfig = new DummyCFamilyRulesConfig("any")
+                .AddRule("rule1", isActive: true)
+                .AddRule("rule2", isActive: false);
+
+            // 1. Match - active
+            var message = new Message("rule1", "filename", 0, 0, 0, 0, "msg", false, null);
+            MessageHandler.IsIssueForActiveRule(message, rulesConfig).Should().BeTrue();
+
+            // 2. Match - not active
+            message = new Message("rule2", "filename", 0, 0, 0, 0, "msg", false, null);
+            MessageHandler.IsIssueForActiveRule(message, rulesConfig).Should().BeFalse();
+
+            // 3. No match - case-sensitivity
+            message = new Message("RULE1", "filename", 0, 0, 0, 0, "msg", false, null);
+            MessageHandler.IsIssueForActiveRule(message, rulesConfig).Should().BeFalse();
+
+            // 4. No match
+            message = new Message("xxx", "filename", 0, 0, 0, 0, "msg", false, null);
+            MessageHandler.IsIssueForActiveRule(message, rulesConfig).Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void HandleMessage_IssueProcessing_InactiveRulesAreIgnored_ActiveRulesAreNotIgnored()
```

Comment:
Added. This is a simplified version of the `TriggerAnalysisAsync_IssuesForInactiveRulesAreNotStreamed` test - it doesn't need to set up the request factory or handle analysis notifications.